packages/swarm-components/src/screenRenderer.js

```diff
@@ -0,0 +1,106 @@
+import Hapi from 'hapi';
+import Path from 'path';
+import inert from 'inert';
+
+import puppeteer from 'puppeteer';
+
+import ReactDOMServer from 'react-dom/server';
+
+class screenRenderer {
+	constructor(config) {
+		this.config = { verbose: false, port: 4000, padding: '1em', ...config };
+
+		this.routeIndex = 0;
+	}
+
+	log(...params) {
+		if (this.config.verbose) {
+			console.log(...params);
+		}
+	}
+
+	async init() {
+		const { port, staticPath } = this.config;
+
+		this.browser = await puppeteer.launch();
+		this.log('Browser is running');
+
+		this.server = Hapi.server({
+			port,
+		});
+
+		await this.server.register(inert);
+
+		// serving static files
+		this.server.route({
+			method: 'GET',
+			path: '/static/{param*}',
+			handler: {
+				directory: {
+					path: Path.join(__dirname, staticPath),
+				},
+			},
+		});
+
+		await this.server.start();
+		this.log('Server running on %s', this.server.info.uri);
+	}
+
+	createRoute(slug, element) {
+		const links = this.config.stylesheets
+			.map(
+				stylesheet =>
+					`<link
+						rel="stylesheet"
+						type="text/css"
+						href="/static/${stylesheet}" />`
+			)
+			.join('\n');
+
+		return {
+			method: 'GET',
+			path: `/${slug}`,
+			handler: () => `<html>
+								<head>
+									${links}
+								</head>
+								<body style="padding: ${this.config.padding}">
+								${ReactDOMServer.renderToStaticMarkup(element)}
+								</body>
+							</html>`,
+		};
+	}
+
+	async stop() {
+		await this.browser.close();
+		return this.server.stop();
+	}
+
+	async screenshot(element) {
+		const page = await this.browser.newPage();
+		if (this.config.viewport) {
```

Comment:
feel free to ignore for now, but was wondering if the `screenshot` method should take an optional config override, at least for the viewport, so that elements with viewport-responsive styles can be rendered at different widths in tests.

Also, instead of testing for the existence of `this.config.viewport`, I think it would be better to define a default width and always call `page.setViewport`, since an 'undefined' viewport is ambiguous (presumably puppeteer has an internal default?)

e.g.

```js
async screenshot(element, shotConfig) {
  const page = await this.browser.newPage();
  page.setViewport(shotConfig && shotConfig.viewport || this.config.viewport);
  ...
}
```

Reply:
Yes, puppeteer has internal default - we can standardize on something for this purpose or just explicitly use puppeteer's current default.